src/js-form-validator.ts

```diff
@@ -313,10 +313,8 @@ export class FormValidator {
       case 'phone':
         if (!allowEmpty && !value.length) return -1;
         let vals = value.split('x');
-        for (let i = 0, l = vals.length; i < l; i++)
-          vals[i] = vals[i].replace(/\D/g, '');
         value = vals.join('x').replace(/^1/, ''); // remove leading 1 if added, only for US
-        status = /^[0-9x]{7,16}$/.test(value) ? 1 : 0;
+        status = /^[0-9\-\+\s\(\)]{7,16}$/.test(value) ? 1 : 0;
```

Comment:
>using the compiler for the sake of error finding

No, there should be absolutely no errors in this code, as it should be checked before it passes a PR.

The point of not using the compiled files is so we can use them in our TypeScript projects. Without it, then we don't have access to interfaces and in general, better intellisense and visibility into the module without the need of doing @types.

>The issue that was happening was that since we were stripping out those other characters the entry was able to scatter letters in the entry and it would pass validation

Do you have an example of this? I'm not following how adding any letters besides an x would allow it to pass validation.

I think we should ultimately just update the regular expression a little bit if we find we need to do that and only validate against numbers.